core/src/main/scala/spinal/core/Trait.scala

```diff
@@ -510,6 +510,11 @@ trait Nameable extends OwnableRef with ContextUser{
   def setName(name : String) : this.type = setName(name, false)
   def setName(name: String, weak: Boolean): this.type = setName(name, if(weak) USER_WEAK else USER_SET)
   def setName(name: String, namePriority: Byte): this.type = {
+    val words = "[A-Za-z_]*\\w*".r
```

Comment:
I'm a bit worry about having a regex being compiled each time there is a setname (speed / overhead)

Also, having it being checked in the Nameable trait is problematic :
- Nameable isn't only used for verilog generation, but as a general way to construct names, which may have different rules
- This check isn't applied to other naming primitives.

I think the best is to add the check in the name allocation phase : 
https://github.com/SpinalHDL/SpinalHDL/blob/734e31103e1731b557a7ec05c743e7bd8a73014a/core/src/main/scala/spinal/core/internals/Phase.scala#L2233

Reply:
> I'm a bit worry about having a regex being compiled each time there is a setname (speed / overhead)

Yes,  I have meet case of  generating illegal Verilog code.   But It's true that is not very common. it's a overhead.

>Nameable isn't only used for verilog generation, but as a general way to construct names, which may have different rules

all right， In that case,  guarantee user themselves  ,  let  me remove this  code